roles/puppet_repositories/tasks/debian.yml

```diff
@@ -1,10 +1,16 @@
 ---
-- name: 'Install GPG package'
+- name: Install GPG package
   package:
-    name: 'gpg'
+    name: gpg
     state: present
     update_cache: true
 
-- name: 'Install Puppet repository'
+- name: Install Puppet repository
   apt:
-    deb: https://apt.puppet.com/puppet{{ foreman_puppet_repositories_version }}-release-{{ ansible_distribution_release }}.deb
+    deb: "https://apt.puppet.com/puppet{{ foreman_puppet_repositories_version }}-release-{{ ansible_distribution_release }}.deb"
+    state: present
+  register: puppet_apt_deb_install
+
+- name: Update apt cache if necessary
+  apt: update_cache=yes force_apt_get=yes
+  when: puppet_apt_deb_install.changed
```

Comment:
this should be probably a handler, notified by the previous task?

Reply:
Totally agree; however, there is an issue with using a handler here. Handlers run at the end of a play unless we use the `meta: flush_handlers` sometime after the repo is added and before the installer is called. Generally speaking, I think most people using this repo are adding the puppet repos and then calling the installer. If we were to flush_handlers between the two, we might force a bunch of other handlers to run that the developer did not intend to be run before the play completed.

The foreman-installer must not update the repo cache before installing the package. Nonetheless, this change will ensure we update the repo prior to using the installer.

Reply:
Ah, yes of course!

Reply:
I just saw the pipeline failed due to the lint error. I've added `skip_ansible_lint` tag. Let me know if you think this is overkill or there is a better way to handle this. :)